performance-tracking/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/AnalyticsBroadcaster.java

```diff
@@ -52,9 +58,17 @@ private static void sendEvent(
    * Construct instance of AnalyticsBroadcaster that holds a {@link WeakReference} to the context.
    *
    * @param context Context that will be used for broadcasting
+   * @param blacklist variable list of URL strings that will be used to blacklist broadcasts.
    */
-  AnalyticsBroadcaster(@NonNull Context context) {
+  AnalyticsBroadcaster(@NonNull Context context, String... blacklist) {
     this.context = new WeakReference<>(context);
+    this.blacklist = new ArrayList<>(blacklist.length);
+    for (String url : blacklist) {
+      try {
+        this.blacklist.add(new URL(url));
+      } catch (MalformedURLException ignored) {
```

Comment:
i think it's ok to silently fail, because the blacklist of urls is purely internal, the can not do anything about it. If the url fails to parse it's either a mistake in the SDK OR it is a configuration mistake in the analytics SDK, in which case the analytics SDK should log about it.